File: benchmark/c/bench_bubble_sort.c
int main(void) {
  int a[1000];
  int n = 1000;
  int i = 0;

  while (i < n) {
    a[i] = n - i;
    i++;
  }

  for (i = 0; i < n - 1; i++) {
    for (int j = 0; j < n - 1 - i; j++) {
      if (a[j] > a[j + 1]) {
        int temp = a[j];
        a[j] = a[j + 1];
        a[j + 1] = temp;
      }
    }
  }
  return 0;
}

File: benchmark/python/bench_bubble_sort.py
N = 1000


def bubble_sort(values: list[int]) -> None:
  for i in range(N - 1):
    for j in range(N - 1 - i):
      if values[j] > values[j + 1]:
        temp = values[j]
        values[j] = values[j + 1]
        values[j + 1] = temp


def main() -> None:
  a = [0] * N
  i = 0
  while i < N:
    a[i] = N - i
    i += 1
  bubble_sort(a)


if __name__ == "__main__":
  main()
